File: sample/src/main/java/contacts/sample/cheatsheet/data/java/UpdateDataActivity.java
package contacts.sample.cheatsheet.data.java;

import android.app.Activity;

import java.util.Set;

import contacts.core.ContactsFactory;
import contacts.core.Fields;
import contacts.core.data.DataUpdate;
import contacts.core.entities.*;

public class UpdateDataActivity extends Activity {

    DataUpdate.Result updateDataSet(Set<ExistingDataEntity> dataSet) {
        return ContactsFactory.create(this).data().update().data(dataSet).commit();
    }

    DataUpdate.Result updateEmailAndPhone(Email email, Phone phone) {
        MutableEmail mutableEmail = email.mutableCopy();
        mutableEmail.setAddress("myemail@email.com");

        MutablePhone mutablePhone = phone.mutableCopy();
        mutablePhone.setNumber("(555) 555-5555");

        return ContactsFactory.create(this)
                .data()
                .update()
                .data(mutableEmail, mutablePhone)
                .commit();
    }

    DataUpdate.Result updateOnlyMiddleName(MutableName changedName) {
        return ContactsFactory.create(this)
                .data()
                .update()
                .data(changedName)
                .include(Fields.Name.MiddleName)
                .commit();
    }
}

File: sample/src/main/java/contacts/sample/cheatsheet/data/kotlin/UpdateDataActivity.kt
package contacts.sample.cheatsheet.data.kotlin

import android.app.Activity
import contacts.core.Contacts
import contacts.core.Fields
import contacts.core.data.DataUpdate
import contacts.core.entities.*

class UpdateDataActivity : Activity() {

    fun updateDataSet(dataSet: Set<ExistingDataEntity>): DataUpdate.Result =
        Contacts(this).data().update().data(dataSet).commit()

    fun updateEmailAndPhone(email: Email, phone: Phone): DataUpdate.Result = Contacts(this)
        .data()
        .update()
        .data(
            email.mutableCopy {
                address = "myemail@email.com"
            },
            phone.mutableCopy {
                number = "(555) 555-5555"
            }
        )
        .commit()

    fun updateOnlyMiddleName(changedName: MutableName): DataUpdate.Result = Contacts(this)
        .data()
        .update()
        .data(changedName)
        .include(Fields.Name.MiddleName)
        .commit()
}